aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferStatusUpdater.java

```diff
@@ -147,6 +148,11 @@ synchronized void removeTransfer(int id) {
     synchronized void removeTransferRecordFromDB(final int id) {
         S3ClientReference.remove(id);
         dbUtil.deleteTransferRecords(id);
+        // Remove temporary file
+        if (dbUtil.getTransferById(id).file.startsWith("aws-s3")) {
+            String path = dbUtil.getRecordUri(id).getPath();
```

Comment:
I think this is getting a reference to a `Uri` associated with the database entry for this particular upload, not the temporary `File`.  `dbUtil.getTransferById(id).file` should give you the path to the File.

Also, I believe you need to obtain the `File` path before `dbUtil.deleteTransferRecords(id);` is called.   Otherwise, the transfer record won't be there anymore :)

Lastly, `getTransferById(id)` may return null, so it would be good to add a null check after obtaining it.